File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
a, b = map(int, input().split())

# 최대 공약수
def gcd(a, b):
    # a에 더 큰수를 담는다.
    if a < b:
        tmp = a
        a = b
        b = tmp
    # 유클리드 알고리즘
    while b != 0:
        r = a % b
        a = b
        b = r
    return a
gcd = gcd(a, b)
print(gcd)

# 최소 공배수
def lcm(a, b, gcd):
    return (a * b) // gcd 

print(lcm(a, b, gcd))

File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
import java.io.*;
import java.util.*;

public class Main {
    public static int gcd(int a, int b) {
        while (b != 0) {
            int r = a % b;
            a = b;
            b = r;
        }
        return a;
    }
    
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        
        int a = Integer.parseInt(st.nextToken());
        int b = Integer.parseInt(st.nextToken());
        
        int gcd;
        int lcm;
        
        gcd = gcd(a, b);
        lcm = (a * b) / gcd;
        
        System.out.println(gcd);
        System.out.println(lcm);
    }
}



    